### ISSUE TITLE
Enyaq NO_MOD1_4_VEHICLES error - Could not obtain GDC token

### ISSUE THREAD

**Issue by fm4991** [skodaconnect/homeassistant-myskoda#494]
since a few days my homeassistant said myskoda unavailable. Reload didn't help.
So i deleted the Hub for my car and tried to set it up again.

But i get an Unkown Error during setup.
the only thing i find in the logs is:

![image](https://github.com/user-attachments/assets/5139664e-bb8c-4d80-99ca-dea738fc114d)

**Comment by prvakt** [comment]
please update to latest version, seems like you are running older version where MQTT was failing because of change made by MySkoda - this was fixed in `1.15.3` and latest version is `1.17.2`


just fyi this problem is still not completely fixed - seems like MySkoda implemented some kind of restriction for the UUID and some UUIDs are working and some not.. so can happen that you are good to go during first connect, sometimes second or third but sometimes it fails multiple times in row which may result in increased reconnect backoff to a minute or two..   

**_NOTE:_** we are using random uuid for connection to MQTT as this was done by myskoda app as well, but probably they have some range of allowed UUIDs - until we will not find how they build this UUID, we can just generate a random one and retry with another one if it fails

**Comment by fm4991** [comment]
Hey @prvakt,

thanks for the fast answer. Yeah seems like i did a manual install and never updated it 🙈🫣

but now i have another problem.
I add my Account this seems to be successfull.
But there are no entities created.

**Comment by prvakt** [comment]
> Hey @prvakt,
> 
> thanks for the fast answer. Yeah seems like i did a manual install and never updated it 🙈🫣
> 
> but now i have another problem. I add my Account this seems to be successfull. But there are no entities created.

If you have a CitigoE iV, then please use `1.16.1`. There is a problem with one endpoint in `1.17.x` which doesn't work for Citigo and we are fixing this problem right now. 

If you have another car, maybe you are locked out from MySkoda right now. When creating issue next time, please put some more details at least about your car model and version used (ideally with diagnostics report from MySkoda integration) - we will be able to advise you better

**Comment by fm4991** [comment]
I have a Skoda Enyaq, was using latest 1.17.2 but also tried 1.16.1 now here are the debugging logs from the try with 1.16.1:
[home-assistant_myskoda_2024-12-30T13-08-00.286Z.log](https://github.com/user-attachments/files/18274846/home-assistant_myskoda_2024-12-30T13-08-00.286Z.log)

here the debug log of my try with 1.17.2, when i click on download diagnostics i get an error message from my browser that there is no such file on the server
[home-assistant_myskoda_2024-12-30T13-14-36.395Z.log](https://github.com/user-attachments/files/18274906/home-assistant_myskoda_2024-12-30T13-14-36.395Z.log)

![image](https://github.com/user-attachments/assets/60a2816d-193a-4815-8eaf-8737809ddaea)

**Comment by prvakt** [comment]
you need to go inside your Skoda Enyaq entity and then download diagnostics which is available under device info
![Screenshot 2024-12-24 140802](https://github.com/user-attachments/assets/56c57ed1-93e8-49dd-9928-cdeb1f1858c2)

But please first delete the hub and configure it again.. If no entities are created again , I will need to see a log capturing this as well

**Comment by fm4991** [comment]
I don't have an entity for my enyaq :-( the debug logs i attached before already include that.

**Comment by prvakt** [comment]
> I don't have an entity for my enyaq :-( the debug logs i attached before already include that.

So if you delete the hub and create again with 1.17.x, there is no device created?

**Comment by fm4991** [comment]
no i even delete the whole myskoda integration cleared all existing configurations and redownloaded it with hacs. i create the hub and afterwards nothing is happening.

**Comment by prvakt** [comment]
unfortunately I don't see any error or anything wrong in the log (@WebSpider can you have a look as well please?).. 

Please enable debug logging from the integration page, maybe we can see something in the debug logs

**Comment by fm4991** [comment]
i also can't see any errors in the logs. the logs which i've attached before are already with enabled debug mode on the integration page.

**Comment by prvakt** [comment]
can you confirm, that with MySkoda mobile app you can see your vehicle?

**Comment by fm4991** [comment]
yes i can
![image](https://github.com/user-attachments/assets/e15a4ba8-3883-41df-874b-31e6a756b453)

**Comment by WebSpider** [comment]
I'll have a look as well, from mobile i cant see much wrong, so will have a look on a desktop in a few hrs

**Comment by prvakt** [comment]
Then please enable also API response tracing which you can set via "CONFIGURE" button. 
Once debug logging and API tracing is enabled, restart home assistant and provide me the log - as it will contain sensitive data, you can send it to me via email (`t.prvak@gmail.com`) or via our Discord channel (my nick is `EnergyX`)

@WebSpider: Only thing which I'm missing is there is no connection to MQTT which should be traced right after `IDK Authorization was successful.` and `MySkoda ready.` traces

**Comment by WebSpider** [comment]
Mqtt initialisation is async now iirc, it can lag behind (and thats fine, mqtt is an additional source of data, besides the rest api)

**Comment by fm4991** [comment]
@prvakt i sent the log to you on discord...
the only line which sounds strange to me is this one:
```
2024-12-30 15:44:27.529 DEBUG (MainThread) [myskoda.myskoda] Trace: GET https://mysmob.api.connect.skoda-auto.cz/api/v2/garage?conne - response: 200 (100 bytes) {"vehicles":[],"errors":[{"type":"NO_MOD1_4_VEHICLES","description":"Could not obtain GDC token."}]}
```

**Comment by prvakt** [comment]
> @prvakt i sent the log to you on discord... the only line which sounds strange to me is this one:
> 
> ```
> 2024-12-30 15:44:27.529 DEBUG (MainThread) [myskoda.myskoda] Trace: GET https://mysmob.api.connect.skoda-auto.cz/api/v2/garage?conne - response: 200 (100 bytes) {"vehicles":[],"errors":[{"type":"NO_MOD1_4_VEHICLES","description":"Could not obtain GDC token."}]}
> ```

yep that's actually why your vehicle is not loaded but I have never seen this error before - - any idea @WebSpider ?

**Comment by WebSpider** [comment]
This is an internal error to MySkoda, there are no supported cars linked to your account. Usually this is temporary.

MOD1-4 is the name of the 4 generations connectivity hardware in the car.

**Comment by fm4991** [comment]
Is it worth a try to remove my enyaq from the account and add it again?

**Comment by prvakt** [comment]
> > Is it worth a try to remove my enyaq from the account and add it again?
> 
> Try and login to the Skoda online shop where you can update your subscriptions. If your car doesnt show there, contact Skoda support.

How it is possible that he can see and control the car in MySkoda app?

**Comment by fm4991** [comment]
Aha! let's see if they will fix it 😓 
![image](https://github.com/user-attachments/assets/7394a2c2-683d-42c9-9766-2dfda066ecc2)

**Comment by WebSpider** [comment]
> > > Is it worth a try to remove my enyaq from the account and add it again?
> > 
> > Try and login to the Skoda online shop where you can update your subscriptions. If your car doesnt show there, contact Skoda support.
> 
> How it is possible that he can see and control the car in MySkoda app?

Probably caching. We rediscover the VINs upon startup.

**Comment by dvx76** [comment]
@fm4991 is this still an issue?

**Comment by fm4991** [comment]
@dvx76 yes nothing changed… wrote an email to skoda support. Didn‘t get an answer yet.

**Comment by fm4991** [comment]
Hey guys,
It works again. In the meantime my enyaq also dissapeared in the MySkoda App and i wasn’t able to readd it again. Skoda Support didnt really help. 
so i tried every day to add it again and today was the day. Ha integration is also working again.
Thanks for your efforts.
